targets/TARGET_Analog_Devices/TARGET_ADUCM4X50/TARGET_ADUCM4050/bsp/rng/adi_rng.c

```diff
@@ -0,0 +1,796 @@
+
+/*!
+ * @brief  Enables/Disables Buffering for RNG.
+ *
+ * @param[in]   hDevice       Device handle obtained from adi_rng_Open().
+ * @param[in]   bFlag         Flag to specify whether to enable or disable buffering for RNG device.
+ *                            When buffering is enabled, adi_rng_GetRngData returns 32-bit values.
+ *                            When buffering is disabled the API returns 8-bit values.
+ *
+ * @return      Status
+ *                - #ADI_RNG_SUCCESS                    Call completed successfully.
+ *                - #ADI_RNG_BAD_DEV_HANDLE [D]     Invalid device handle parameter.
+ *                - #ADI_RNG_NOT_INITIALIZED [D]    Device has not been initialized for use, see #adi_rng_Open().
+ *
+ * @sa        adi_rng_Open().
+ * @sa        adi_rng_RegisterCallback().
+ * @sa        adi_rng_GetRngData().
+ */
+ADI_RNG_RESULT adi_rng_EnableBuffering (ADI_RNG_HANDLE const hDevice, bool const bFlag)
+{
+    ADI_RNG_DEV_TYPE *pDevice = (ADI_RNG_DEV_TYPE*)hDevice;
+    ADI_INT_STATUS_ALLOC();
+
+#ifdef ADI_DEBUG
+    if (ADI_RNG_INVALID_HANDLE(pDevice)) {
+        return ADI_RNG_BAD_DEV_HANDLE;
+    }
+
+    if (NULL == pDevice->pData) {
+        return ADI_RNG_NOT_INITIALIZED;
+    }
+#endif
+
+    ADI_ENTER_CRITICAL_REGION();
+    if (true == bFlag) {
+        pDevice->pRNG->CTL &= (uint16_t)~(BITM_RNG_CTL_SINGLE);
+    } else {
+        pDevice->pRNG->CTL |= BITM_RNG_CTL_SINGLE;
+    }
+    ADI_EXIT_CRITICAL_REGION();
+
+    return ADI_RNG_SUCCESS;
+}
+
+/*!
+ * @brief  Sets the reload and prescale value for the sample counter.
+ *         The Sample Length will be nLenReload*2^nLenPrescaler.
+ *
+ * @param[in]   hDevice        Device handle obtained from adi_rng_Open().
+ * @param[in]   nLenPrescaler  Prescaler value for the sample counter (0-10).
+ * @param[in]   nLenReload     Reload value for the sample counter (0-4095)
+ *
+ * @return      Status
+ *                - #ADI_RNG_SUCCESS                    Call completed successfully.
+ *                - #ADI_RNG_BAD_DEV_HANDLE [D]     Invalid device handle parameter.
+ *                - #ADI_RNG_NOT_INITIALIZED [D]    Device has not been initialized for use, see #adi_rng_Open().
+ *
+ * @sa        adi_rng_Open().
+ * @sa        adi_rng_RegisterCallback().
+ */
+ADI_RNG_RESULT adi_rng_SetSampleLen (
+                                     ADI_RNG_HANDLE const hDevice,
+                                     uint16_t       const nLenPrescaler,
+                                     uint16_t       const nLenReload
+                                     )
+{
+    ADI_RNG_DEV_TYPE *pDevice = (ADI_RNG_DEV_TYPE*)hDevice;
+    ADI_INT_STATUS_ALLOC();
+
+#ifdef ADI_DEBUG
+    if (ADI_RNG_INVALID_HANDLE(pDevice)){
+        return ADI_RNG_BAD_DEV_HANDLE;
+    }
+
+    if (NULL == pDevice->pData) {
+        return ADI_RNG_NOT_INITIALIZED;
+    }
+
+    if (   (nLenPrescaler > 10u)
+        || ((0u == nLenPrescaler) && (0u == nLenReload))
+        || (nLenReload > 4095u)) {
```

Comment:
Why is `nLenReload == 0` not generally forbidden here, regardless of `nLenPrescaler`? In this case, it seems that the resulting CRC sample counter value would be `nLenReload * 2^nLenPrescaler = 0`, independently on the value of `nLenPrescaler`.

Reply:
This code is for debugging purposes only and it is disabled in the build.

Reply:
Ok. It still seems that this check is not accurate, but given that it's debug-only and the values of `nLenPrescaler` and `nLenReload` we're using this function with are hardcoded, I'm ok with leaving it as is.